public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts

```diff
@@ -445,6 +428,20 @@ export class CloudWatchLogsQueryRunner extends CloudWatchRequest {
 
     return getLogGroupFieldsResponse;
   }
+
+  private filterQuery(hasDefaultLogGroups: boolean) {
+    return (query: CloudWatchLogsQuery) => {
+      const hasMissingLegacyLogGroupNames = !query.logGroupNames?.length && !hasDefaultLogGroups;
```

Comment:
@sunker I had some trouble getting the information for `this.defaultLogGroups` here. Kind of recall that the context of `this` isn't going to be the same as the caller which is why.
I used a closure to pass that information, let me know if you have a better way to get that info.

Reply:
Hmm think it will be executed in the same scope and then you should not need the closure? 

Anyway, think the default log groups should have been applied to the logs query before the datasource.query method is being called. If so, you don't need to take this.defaultLogsGroups into consideration at this point - you only need to look at query.logGroupNames.

Reply:
Yeah I mean, in the [LogsQueryEditor](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx#L82), query.logGroupNames is set to defaultLogsGroups (if logGroupNames are empty), so it should be enough to check length of query.logGroupNames here. This is not possible to capture in a unit test since you'd need to involve the QueryEditor too. 

Anyway, the way default queries are applied are about to change soon (next week I think) when Ida picks up [this](https://github.com/grafana/grafana/issues/58205) issue. When this is implemented, we no longer have to call `onChange` from within the query editor to set default values - we just define them in the data source file, and then defaults will be applied before the query editor is rendered the first time and before the datasource.query method is called the first time. If we just make sure then that `query.logGroupNames = defaultLogsGroups`, we don't need to bother about this.defaultLogsGroups in this filter method. So I think you can ignore the failing test for now and remove it.